rudder-agent/SPECS/rudder-agent.spec

```diff
@@ -301,8 +301,7 @@ TRY_CFLAGS="-fPIE -fstack-protector"
 
 FLAG_TEST_FILE=`mktemp /tmp/hardening.XXXXXX`
 echo "void main() {}" > "${FLAG_TEST_FILE}.c"
-gcc ${TRY_CFLAGS} ${TRY_LDFLAGS} -o "${FLAG_TEST_FILE}" "${FLAG_TEST_FILE}.c" 2>/dev/null
-if [ $? -eq 0 ]
+if [ $(gcc ${TRY_CFLAGS} ${TRY_LDFLAGS} -o "${FLAG_TEST_FILE}" "${FLAG_TEST_FILE}.c" >/dev/null 2>&1; echo $?) -eq 0 ]
```

Comment:
I think the problem is set -e and this scheme will fail with set -e too.
Instead use: if gcc ..... ; then true; else ... ; fi